docs/building-a-form/about-the-json-schema-standard-and-react-jsonschema-form.md

```diff
@@ -0,0 +1,298 @@
+
+```
+{
+  type: 'array',
+  items: {
+    type: 'boolean'
+  }
+}
+```
+
+Many libraries implement the JSON Schema specification and let you validate that an object matches a given schema. The US Forms System uses [ajv](https://www.npmjs.com/package/ajv) for unit tests and [jsonschema](https://www.npmjs.com/package/jsonschema) in the schemaform code. For more examples of JSON Schema projects, see the [JSON Schema site example list](http://json-schema.org/examples.html).
+
+### Understanding react-jsonschema-form (rjsf)
+
+[react-jsonschema-form](https://github.com/mozilla-services/react-jsonschema-form), or *rjsf*, generates a form from a JSON Schema, in addition to other UI information. To generate a form, react-jsonschema-form steps through the schema depth and renders different React components based on the type of data each property in the schema represents.
+
+##### About the `Form` component
+
+At the top level, rjsf uses a `Form` component to take the schema inputs and render a hierarchy of components for each field rendered on the form:
+
+- *Fields* generally match the `type` attribute in a schema. There are object fields, array fields, number fields, boolean fields, and string fields. Except for arrays and objects, the fields render a label (via `FieldTemplate`) and a widget.
+- A *widget* is the html input element that accepts data from the user. The schemaform uses `text`, `email`, `checkbox`, `radio`, `select`, and `textarea`. While there are many widgets provided by rjsf, the defaults are overwritten with these versions.
+
+##### Example schema
+
+```
+{
+  type: 'string'
+}
+```
+
+##### Resulting hierarchy
+
+The two `Field` components determine which fields and widgets to render. `SchemaField` uses the two schemas the library accepts, `schema` and `uiSchema`, to determine what other `Field` component to render. The example chose `StringField` because the schema type was `string`. The `StringField` component then rendered `TextWidget`, based on `schema` and `uiSchema`, because the only information provided was that the field is a string (the default widget type).
+
+```
+<SchemaField>
+  <StringField>
+    <FieldTemplate>
+      <TextWidget/>
+    </FieldTemplate>
+  </StringField>
+</SchemaField>
+```
+
+##### Example schema
+
+```
+{
+  type: 'string',
+  enum: ['first', 'second', 'third']
+}
+```
+
+##### Resulting hierarchy
+
+The hierarchy for this field uses `SelectWidget` instead of `TextWidget`, because `StringField` detected the `enum` property in the schema.
+
+```
+<SchemaField>
+  <StringField>
+    <FieldTemplate>
+      <SelectWidget/>
+    </FieldTemplate>
+  </StringField>
+</SchemaField>
+```
+
+To override rules in `uiSchema`, specify a `ui:widget` property set to `text`, `email`, `checkbox`, or your own custom widget. The `ui:field` property can specify a specific or custom field.
+
+For `object` and `array` schema types, a field component does something other than figure out what widgets to render.
```

Comment:
This transition is pretty abrupt, and is introducing a new concept. I think it needs it's own header, and more context. I might rewrite this to be:

"While in most cases a field component is responsible for rendering a label and a widget, for `object` and `array` schema types, the field component renders additional field components for each of the elements they contain."

Reply:
I've taken your suggestion verbatim.

As for needing a header here, I totally agree, and that goes for pretty much the entire document. It needs better organization. I'd like to do that in a separate pull request, along with some pairing with you and Dave to talk this out. I opened an issue about this in https://github.com/usds/us-forms-system/issues/139.